src/Storage/Storage.Management/File/NewAzureStorageShare.cs

```diff
@@ -112,11 +112,32 @@ public string AccessTier
 
         [Parameter(Mandatory = false, HelpMessage = "Create a snapshot of existing share with same name.")]
         public SwitchParameter Snapshot { get; set; }
+        
+        [Parameter(Mandatory = false,
+            HelpMessage = "Sets protocols for file shares. It cannot be changed after file share creation. Possible values include: 'SMB', 'NFS'")]
+        [ValidateSet(EnabledProtocols.NFS,
+            EnabledProtocols.SMB,
+            IgnoreCase = true)]
+        public string EnabledProtocol { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "Sets reduction of the access rights for the remote superuser. Possible values include: 'NoRootSquash', 'RootSquash', 'AllSquash'")]
+        [ValidateSet(RootSquashType.NoRootSquash,
+            RootSquashType.RootSquash,
+            RootSquashType.AllSquash,
+            IgnoreCase = true)]
+        public string RootSquash { get; set; }
```

Comment:
if you are very unlikely to extend RootSquash types, it is ok to use validate set. If not, argument completer will be better.

Reply:
same as above